timefold-solver/src/test/java/com/baeldung/timefoldsolver/ShiftScheduleSolverUnitTest.java

```diff
@@ -0,0 +1,57 @@
+package com.baeldung.timefoldsolver;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ai.timefold.solver.core.api.solver.Solver;
+import ai.timefold.solver.core.api.solver.SolverFactory;
+import ai.timefold.solver.core.config.solver.SolverConfig;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+public class ShiftScheduleSolverUnitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ShiftScheduleSolverUnitTest.class);
+
+    @Test
+    public void solve() {
+        SolverFactory<ShiftSchedule> solverFactory = SolverFactory.create(new SolverConfig().withSolutionClass(ShiftSchedule.class)
+            .withEntityClasses(Shift.class)
+            .withConstraintProviderClass(ShiftScheduleConstraintProvider.class)
+            // The solver runs only for 2 seconds on this tiny dataset.
+            // It's recommended to run for at least 5 minutes ("5m") on large datasets.
+            .withTerminationSpentLimit(Duration.ofSeconds(2)));
+        Solver<ShiftSchedule> solver = solverFactory.buildSolver();
+
+        ShiftSchedule problem = loadProblem();
+        ShiftSchedule solution = solver.solve(problem);
```

Comment:
Sorry I missed this too, can we actually assert on the solution, rather than just printing it, if it is deterministic?

Reply:
If we keep `.withTerminationSpentLimit(Duration.ofSeconds(2)));` it is possible that extremely slow working nodes with a large stop-the-world GC might **randomly fail** the unit test. But in the unit test we know the optimal score, so we can safely change it to this:

```
// For this dataset, we know the optimal score in advance (which is normally not the case).
            // So we can use .withBestScoreLimit() instead of .withTerminationSpentLimit().
            .withTerminationConfig(new TerminationConfig().withBestScoreLimit("0hard/0soft")));
            ```